### PR TITLE
QuantumUtilis Docstrings and Tests 

### PR DESCRIPTION
**Pull request by Fe-r-oz**
Resolving https://github.com/aarontrowbridge/QuantumCollocation.jl/issues/102

### CODE REVIEW

**Comment by Fe-r-oz** [comment]
I think the testing section is almost completed. Tests it locally as well! 

Screenshot_select-area_20240601152514

Please let me know your comments having a brief look over the test file. Thanks!

File: docs/literate/man/utils.jl [hunk 1]
```diff
@@ -1,13 +1,111 @@
 # # Utilities
 
 # ## Gates
+# ```@docs
```
Comment (Fe-r-oz) [inline docs/literate/man/utils.jl:4]: This 
```
│ ```@docs
│ QuantumUtils.GATES
│ ```
```
needs to be removed. It's causing the CI doc error.

Comment (Fe-r-oz) [reply docs/literate/man/utils.jl:4]: Applied the Fix with recent commit.

**Comment by andgoldschmidt** [comment]
Hi @Fe-r-oz, closing this PR until it is ready to be reviewed by a maintainer. In the meantime, please use the issues page to keep up our discussion!

**Comment by Fe-r-oz** [comment]
Dear @aarontrowbridge, It is my sincere request to please review this PR. I would be grateful! Thank you.

**Comment by Fe-r-oz** [comment]
Dear @andgoldschmidt, with regards to this PR #111 , Kindly please open this one because it's been ready for review!

I am using issue page for discussion for other PR, not this one!  Thank you very much!

**Review by aarontrowbridge — COMMENTED** [review]
I think we would like to see these docstrings at least include the argument signatures, and ideally a short description. Copilot should be able to do this pretty much automatically.

**Comment by Fe-r-oz** [comment]
Hi, @aarontrowbridge  and @andgoldschmidt,

I provided short descriptions in the docstring along with argument signatures for almost all the docstrings. Sorry, I skipped the signature for two namely kronecker product and Gates.

If the copilot can automatically provide docstrings, Should I remove them and only submit the `quantum_utils_test`? I would be delighted to incorporate changes here. 

I wasn't sure that copolit can automatically handle documentation, that's why I added them as my assumption was that I had to write both documentation along with tests.

 I will wait for the review and try to incorporate the changes.

**Pull request closed without merging**